README.md

```diff
@@ -1,3 +1,360 @@
+   wheel - once we have a better idea of what calls we need to make, it may make sense to split some of
+   `composer/composer` out into separate libraries we can consume; although that might be a huge undertaking in itself,
+   so is really just a "strech goal".
+ * There may be multiple versions of php on a system. The version of php invoking `npecl` would be assumed, and its
+   corresponding `phpize` tool would be used.
+ * `npecl` would probably need to be run with `sudo` access (so it can `make install` to root-owned paths). Care should
+   be taken to ensure only the minimal amount is run with elevated privileges, so we don't accidentally give root to a
+   malicious PECL package. Ideally, `npecl` would run without `sudo`, and just prompt for access, if possible.
+
+### CLI commands
+
+#### `install {ext-name}{?:version-constraint}{?@dev-branch-name}`
+
+Installs the requested package. The package may be requested with or without `ext-` prefix. `npecl install ext-xdebug`
+and `npecl install xdebug` would be equivalent.
+
+If `version-constraint` is given, try to install that version if it matches the allowed versions.
+
+On Windows this downloads the correct DLL (attached as file to the release tag), if available. Otherwise, it will
+download the source and compile the extension.
+
+It will then attempt to create a 20-{extension_name}.ini file with `extension={extension_name}` in the directory
+returned by `php-config –ini-dir`, the name that is used is the one from the "name" element in the metadata file.
+
+If `@dev-branch-name` is given, try to install from the branch called "branch-name", for example use `@dev-master` to
+compile from your master branch. As branches would usually not have binaries, this would not work for Windows.
+
+You can't use both a `version-constraint` and `@dev-branch-name` at the same time.
+
+Each step should run with as few privileges as possible.
+
+#### `build {ext-name}{?:version-constraint}{?@dev-branch-name}`
+
+Same behaviour as install, but only compiles (or downloads if it's Windows).
+
+#### `changelog {?version}`
+
+Shows the release notes of the version it was going to install, or from the specific version, if given.
+
+#### `download {ext-name}{?:version-constraint}{?@dev-branch-name}`
+
+Same behaviour as build, but puts the files in a local directory for manual building and installation.
+
+#### `info`
+
+Shows the description from the metadata file
+
+#### `list`
+
+Shows all available commands
+
+#### `self-update`
+
+Attempts to upgrade the npecl tool itself
+
+#### `show`
+
+Shows all installed extensions available with the PHP version in the path, including their versions. This includes
+*all* loaded PHP extensions, and not just npecl ones.
+
+#### `verify {?version-constraint}`
+
+Shows the signature of the person who signed the version npecl would install, or from the specific version, if given.
```

Comment:
@Seldaek / @naderman Does Composer support fetching the signature of a release from somewhere at the moment? I suppose, we'd be looking for a signed tag in a git repo, perhaps, although, we'd have to verify that the contents of the downloaded zip/DLL matches the git repo content. Not entirely sure what flow @derickr envisioned here? :thinking:

Reply:
Nope there is no infra for that yet in GitHubDriver nor GitDriver AFAIK.

Reply:
thanks for confirming